src/trufflesom/primitives/Primitives.java

```diff
@@ -221,19 +222,16 @@ private static SInvokable constructPrimitive(final SSymbol signature,
       final Specializer<ExpressionNode, SSymbol> specializer,
       final StructuralProbe<SSymbol, SClass, SInvokable, Field, Variable> probe) {
     CompilerAsserts.neverPartOfCompilation("This is only executed during bootstrapping.");
+
     final int numArgs = signature.getNumberOfSignatureArguments();
 
-    MethodGenerationContext mgen = new MethodGenerationContext(probe);
+    // args is needed to communicate the number of arguments to the constructor
     ExpressionNode[] args = new ExpressionNode[numArgs];
```

Comment:
Will this array always be empty? Shouldn't there be a way to provide `specializer` only the number of arguments, and have it create the empty array - which I assume it may fill?

Reply:
Didn't think it'd be a performance issue, just a code clarity one if it didn't entail massive changes. Thanks for the explanation